pandas/core/arrays/datetimes.py

```diff
@@ -376,6 +377,16 @@ def _resolution(self):
     # ----------------------------------------------------------------
     # Array-like Methods
 
+    def __array__(self, dtype=None):
+        if is_object_dtype(dtype):
+            return np.array(list(self), dtype=object)
+        elif is_int64_dtype(dtype):
+            return self.asi8
+
+        # TODO: warn that dtype is not used?
+        #  warn that conversion may be lossy?
+        return self._data.view(np.ndarray)  # follow Index.__array__
```

Comment:
am a little bit hesistant to make this change as we should make this more generally rather than piecemeal (e.g. not just for DTI), but for all Index types.

Reply:
I think you're right that we do need to take a close look at all of the `__array__` methods (also `view` and `astype`), and we will as the DTA/TDA sequence of PRs progresses.  Most of that is out of scope for this PR.

Reply:
@jreback Note that this is not Index, but Array. Index is correctly converting to M8[ns], while on Array it is currently actually giving object dtype, which I would say is a bug.

But you are right that this should be fixed for all Array classes